static/js/settings_users.js

```diff
@@ -511,86 +511,93 @@ function handle_reactivation(tbody, status_field) {
+    function submit_user_details() {
+        const role = Number.parseInt($("#user-role-select").val().trim(), 10);
+        const full_name = $("#edit-user-form").find("input[name='full_name']");
+        const profile_data = get_human_profile_data(fields_user_pills);
 
-            const element = "#edit-user-form .custom-profile-field-form";
-            $(element).html("");
-            settings_account.append_custom_profile_fields(element, user_id);
-            settings_account.initialize_custom_date_type_fields(element);
-            fields_user_pills = settings_account.initialize_custom_user_type_fields(
-                element,
-                user_id,
-                true,
-                false,
-            );
-        }
+        const url = "/json/users/" + encodeURIComponent(user_id);
+        const data = {
+            full_name: JSON.stringify(full_name.val()),
+            role: JSON.stringify(role),
+            profile_data: JSON.stringify(profile_data),
+        };
 
-        function submit_user_details() {
-            const role = Number.parseInt($("#user-role-select").val().trim(), 10);
-            const full_name = $("#edit-user-form").find("input[name='full_name']");
-            const profile_data = get_human_profile_data(fields_user_pills);
+        settings_ui.do_settings_change(channel.patch, url, data, status_field);
+        overlays.close_modal("#dialog_widget_modal");
+    }
 
-            const url = "/json/users/" + encodeURIComponent(user_id);
-            const data = {
-                full_name: JSON.stringify(full_name.val()),
-                role: JSON.stringify(role),
-                profile_data: JSON.stringify(profile_data),
-            };
+    dialog_widget.launch({
+        html_heading: $t_html({defaultMessage: "Change user info and roles"}),
+        parent: modal_parent,
+        html_body,
+        on_click: submit_user_details,
+        post_render: set_role_dropdown_and_fields_user_pills,
+        fade: true,
+    });
+}
 
-            settings_ui.do_settings_change(channel.patch, url, data, status_field);
-            overlays.close_modal("#dialog_widget_modal");
-        }
+function handle_human_form(tbody, status_field) {
+    tbody.on("click", ".open-user-form", (e) => {
+        e.stopPropagation();
+        e.preventDefault();
+        const user_id = Number.parseInt($(e.currentTarget).attr("data-user-id"), 10);
+        show_user_settings(user_id, status_field);
+    });
+}
 
-        dialog_widget.launch({
-            html_heading: $t_html({defaultMessage: "Change user info and roles"}),
-            parent: modal_parent,
-            html_body,
-            on_click: submit_user_details,
-            post_render: set_role_dropdown_and_fields_user_pills,
-            fade: true,
-        });
+export function test(user_uid) {
+    $(() => {
+        const status_field = $("#user-field-status").expectOne();
+        show_user_settings(user_uid, status_field);
     });
 }
```

Comment:
This sort of large-scale reformatting is hard to read/review.

Can you prepare a new first commit for the branch, or perhaps just a new PR if that's easier for you, that just extracts a `show_user_settings` function (adjusting indentation/etc. as appropriate) without making any functional changes?  

That would account for most of the diff here, and be very simple to review/verify was done correctly, and then the additional refactoring/feature changes can be in other highly reviewable commits on top of that.